internal/dump/tar.go

```diff
@@ -13,12 +14,25 @@ import (
 )
 
 func (d *Dumper) dumpTar(ctx context.Context, ch <-chan *restic.Node) (err error) {
-	w := tar.NewWriter(d.w)
+	var w *tar.Writer
```

Comment:
Nit: can be simplified to

```
outer := d.w
if d.compress {
outer = gzip.NewWriter(d.w)
}
w := tar.NewWriter(outer)
```

Reply:
But that would lead to `outer` being an io.Writer which does not have a Close func, so the gzip writer cannot be closed in the deferred func, or am I missing something? I'm new to golang and don't know all the right ways.

Reply:
Ok, found out that a type assertion can be used. Will change.